### PR TITLE
Improve error messages

### PR DESCRIPTION
**Pull request by hamzahrmalik**
Avoid logging localizedDescription for errors which don't conform to `LocalizedError`

some errors might come from underlying libraries
This results is unuseful messages if those libraries don't conform their errors to `LocalizedError`
For example, a NIOSSL error gets logged as 
```
The operation could not be completed. (NIOSSL.NIOSSLError error 0.)
```

We do not get the full underlying error

See the added test case for a further example

### CODE REVIEW

File: Tests/RediStackTests/Helpers/RedisErrorTests.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+@testable import RediStack
+import XCTest
+
+class RedisErrorTests: XCTestCase {
```
Comment (fabianfett) [inline Tests/RediStackTests/Helpers/RedisErrorTests.swift:4]:
```suggestion
final class RedisErrorTests: XCTestCase {
```

**Review by Joannis — APPROVED** [review]

**Comment by Joannis** [comment]
Thanks for the PR! Looks good to me

**Comment by fabianfett** [comment]
@swift-server-bot test this please

**Comment by fabianfett** [comment]
@swift-server-bot test this please

**Comment by fabianfett** [comment]
issue with installing jazzy. retry. @swift-server-bot test this please

**Comment by fabianfett** [comment]
@swift-server-bot test this please

**Pull request merged**